src/components/menu/menu.component.js

```diff
@@ -14,12 +80,18 @@ const Menu = ({ menuType = 'light', children }) => {
         {
           React.Children.map(
             children,
-            (child) => {
+            (child, index) => {
+              const isFirstElement = index === 0;
               return (
                 <StyledMenuItem>{
                   React.cloneElement(
                     child,
-                    { menuType },
+                    {
+                      menuType,
+                      ref: menuItemsRefs.current[index],
+                      isFirstElement,
+                      handleKeyDown: (ev, isOpen) => handleKeyDown(ev, index, isOpen)
```

Comment:
Changing the function each time will cause a re-render. I think we can update the `handleKeyDown` function to use `isOpen` from state then we can memoize the index.
```suggestion
                      handleKeyDown: useCallback(ev => handleKeyDown(ev,index), [index])
```

Reply:
Unresolved

Reply:
Unresolved